rfcs/0000_intelligent_resampling.md

```diff
@@ -170,27 +170,35 @@ _Figure 2_. Case where hypothesis has moved out of object's reference frame.
 
 ### Implementation Strategy
 
-We propose eliminating hypotheses that are more than 10% away from the object's boundary. This percentage threshold is arbitrary but was chosen as a relative distance (rather than a fixed distance like 3 cm) to accommodate objects of different sizes. This elimination step should occur after updating hypotheses but before the next sensing step. 
+The existing `_calculate_evidence_for_new_locations()` method in `hypotheses_displacer.py` already provides a mechanism for handling out-of-reference-frame hypotheses through the `max_match_distance` parameter. 
 
-### Computational Complexity
+1. **Distance-based evidence calculation**: For each hypothesis location, the method finds up to `max_nneighbors` nearest points in the object model using a KDTree
+2. **Custom distance metric**: Instead of simple Euclidean distance, it uses `get_custom_distances()` which considers:
+   - Spatial distance between hypothesis and stored points
+   - Alignment of surface normals
+   - Local curvature properties
 
-A naive approach would find the nearest point and determine if it exceeds the distance threshold. To minimize the computational cost of nearest-point searches, we propose approximating the object model with a convex hull and comparing only against the points that constitute this hull.
+This creates an ellipsoidal region around each hypothesis that takes into account of the local curvature and the hypothesis' surface normal. Currently, if none of the nearest neighbors of the object models are in the ellipsoidal region, the hypothesis receives evidence of -1. We propose to simply extend this to remove these hypotheses. 
```

Comment:
As a very first quick test before going too far down the path of implementing the removal of the hypotheses you can just change that line to -100 or some larger number which would be equivalent to removing the hypothesis. That should show us very quickly how this affects performance. If it is positive then we can add code to actually remove the hypothesis (to save computations when the x_percent_threshold is not used)